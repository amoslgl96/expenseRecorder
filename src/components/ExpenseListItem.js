
import React from 'react';
import { Link } from 'react-router-dom'

import moment from 'moment';

import numeral from 'numeral';
//wiring up edit_expense
//first set up a Link to edit-expense page with the expense' id


const ExpenseListItem=({description,amount,createdAt,id})=>(
    <div>
        <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>
        {numeral(amount/100).format('$0,0.00')} 
        - 
        {moment(createdAt).format("MMMM Do, YYYY")}
        </p>
       
    </div>
)
//export const removeExpense = ({ id } = {}) => ({


export default ExpenseListItem;
//if connect() does not specify any arg
// it just means there are no state values passed into
// expenselistitem, but dispatch is still passed in