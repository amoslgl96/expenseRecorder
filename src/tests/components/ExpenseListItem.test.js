import React from 'react';

import ExpenseListItem from '../../components/ExpenseListItem';

import { shallow } from 'enzyme';


import test_expenses from '../fixtures/expenses'




test('Should return an expenseListItem',
()=>{
    const test_expense=test_expenses[0];
    const wrapper=shallow(<ExpenseListItem {...test_expense}/>);

    expect(wrapper).toMatchSnapshot();
})

