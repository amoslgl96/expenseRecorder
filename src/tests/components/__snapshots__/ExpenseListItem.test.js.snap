// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return an expenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    $500.00
    -
    January 5th, 1970
  </p>
</div>
`;
